### ISSUE TITLE
groupBy with label

### ISSUE THREAD

**Issue by kl3sk** [rob-balfre/svelte-select#484]
The group name is `undefined` when `label` is different than `label` [REPL](https://svelte.dev/repl/0868e6085fba42ccb87e956f9b25cd0b?version=3.52.0)

That work if no override is set [REPL](https://svelte.dev/repl/4f447ad7d75543b285260c3a150ec277?version=3.52.0) (or orverride with `label` value)

**Comment by kl3sk** [comment]
At least a temporary fix can be:

```js

    let createGroupHeaderItem = groupValue => {
        return {
            value: groupValue,
            [label]: groupValue
        }
    }
```

```html
<Select {createGroupHeaderItem }></Select>
```

**Comment by rob-balfre** [comment]
@kl3sk thanks so much for the PR.  Merged and released in `5.0.0-beta.36`
